app/frontend/terrier/tabs.ts

```diff
@@ -31,14 +33,52 @@ export type TabSide = typeof Sides[number]
 
 export type TabContainerState = {
     side: TabSide
-    currentTab? : string
+    currentTab?: string
 }
 
 export class TabContainerPart extends TerrierPart<TabContainerState> {
 
     private tabs = {} as Record<string, TabDefinition>
     changeTabKey = Messages.typedKey<{ tabKey: string }>()
     changeSideKey = Messages.typedKey<{ side: TabSide }>()
+    // Contains the permutation as a list of indexes. A convenience method is defined to reorder an array given the
+    // permutation.
+    static readonly tabReorderedKey = Messages.typedKey<{ permutation: number[] }>()
```

Comment:
This shouldn't be static, because in the case where there are multiple tab containers, you'll want to be able to listen to messages from each tab container individually.

Also I think it should just have an array of the keys in the new order, rather than a permutation array (see my comment on renumberTabs)